### PR TITLE
Change ray version to 1.12.0 in the release branch

### PR DESCRIPTION
**Pull request by jianoaix**
## Why are these changes needed?

Ray 1.12 release

## Related issue number

N/A

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/test_runtime_env.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
 
 def test_get_wheel_filename():
-    ray_version = "2.0.0.dev0"
+    ray_version = "1.12.0rc0"
```
Comment (mwtian) [inline python/ray/tests/test_runtime_env.py:36]: Are these ok to be changed? Let's wait for test results. Maybe they should use an obviously fake version string instead.

**Review by mwtian — APPROVED** [review]

**Review by kfstorm — APPROVED** [review]
Java part LGTM.

File: python/ray/tests/test_runtime_env.py (hunk 1 above)
Comment (naijoaix) [reply python/ray/tests/test_runtime_env.py]: Reverted.

File: python/ray/_private/utils.py [hunk 2]
```diff
@@ -1097,13 +1097,13 @@ def get_wheel_filename(
         sys_platform (str): The platform as returned by sys.platform. Examples:
             "darwin", "linux", "win32"
         ray_version (str): The Ray version as returned by ray.__version__ or
-            `ray --version`.  Examples: "2.0.0.dev0"
+            `ray --version`.  Examples: "1.12.0rc0"
```
[the remark this answers is not in the document]
Comment (naijoaix) [reply python/ray/_private/utils.py]: Thanks for catching this! Reverted.

**Review by jianoaix — COMMENTED** [review]
Thanks for review. Can you take a look again?

File: release/requirements.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # ray
-https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-2.0.0.dev0-cp37-cp37m-manylinux2014_x86_64.whl
+https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-1.12.0rc0-cp37-cp37m-manylinux2014_x86_64.whl
```
Comment (jjyao) [inline release/requirements.txt:2]: This shouldn't be changed I think. @krfricke to confirm.

**Review by jjyao — APPROVED** [review]

File: release/requirements.txt (hunk 3 above)
Comment (krfricke) [reply release/requirements.txt]: Confirmed this shouldn't be changed

**Review by krfricke — APPROVED** [review]
LGTM

**Comment by jjyao** [comment]
There is this error:
```
File "/ray/python/ray/dashboard/modules/serve/sdk.py", line 42, in __init__
--
  | version_error_message="Serve CLI is not supported on the Ray "
  | File "/ray/python/ray/dashboard/modules/dashboard_sdk.py", line 209, in _check_connection_and_version
  | raise RuntimeError(version_error_message)
  | RuntimeError: Serve CLI is not supported on the Ray cluster. Please ensure the cluster is running Ray 1.12 or higher.
  | FAILED


```

File: doc/source/data/dask-on-ray.rst [hunk 4]
```diff
@@ -31,6 +31,8 @@ workload. Using the Dask-on-Ray scheduler, the entire Dask ecosystem can be exec
 
      * - Ray Version
        - Dask Version
+     * - ``1.12.0``
+       - ``2022.2.0``
```
Comment (jjyao) [inline doc/source/data/dask-on-ray.rst:35]: I think this belongs to a different PR?

Comment (naijoaix) [reply doc/source/data/dask-on-ray.rst:35]: Yes, I was trying to cherrypick that PR (merged to master already) but it's messing up this PR. I took it down for now.

**Comment by naijoaix** [comment]
> There is this error:
> 
> ```
> File "/ray/python/ray/dashboard/modules/serve/sdk.py", line 42, in __init__
> --
>   | version_error_message="Serve CLI is not supported on the Ray "
>   | File "/ray/python/ray/dashboard/modules/dashboard_sdk.py", line 209, in _check_connection_and_version
>   | raise RuntimeError(version_error_message)
>   | RuntimeError: Serve CLI is not supported on the Ray cluster. Please ensure the cluster is running Ray 1.12 or higher.
>   | FAILED
> ```

How do we determine the Ray cluster version? It looks this PR has missed some file to update the ray version?

**Comment by shrekris-anyscale** [comment]
Let me take a look.

**Comment by shrekris-anyscale** [comment]
@jjyao and I looked into it. It turns out [the library that compares versions](https://github.com/ray-project/ray/blob/master/dashboard/modules/dashboard_sdk.py#L206-L208) considers "1.12.0rc0" to be less than "1.12," which is the required min version to use the Serve CLI.

To make sure the tests run, we can change [this line](https://github.com/ray-project/ray/blob/master/dashboard/modules/serve/sdk.py#L41) to make min version "1.11.1" since that's higher than all the "1.11" versions but lower than all the "1.12" versions.

**Review by mwtian — APPROVED** [review]

File: dashboard/modules/serve/sdk.py [hunk 5]
```diff
@@ -38,10 +38,10 @@ def __init__(
             headers=headers,
         )
         self._check_connection_and_version(
-            min_version="1.12",
+            min_version="1.11.1",
             version_error_message="Serve CLI is not supported on the Ray "
             "cluster. Please ensure the cluster is "
-            "running Ray 1.12 or higher.",
+            "running Ray 1.11.1 or higher.",
```
[the remark this answers is not in the document]
Comment (jjyao) [reply dashboard/modules/serve/sdk.py:44]: Should be fine. This change will only exist in rc branch. The real 1.12 branch will just work fine without this change.

File: release/requirements.txt [hunk 6]
```diff
@@ -15,4 +15,4 @@ expiringdict
 requests
 pytz
 git+https://github.com/ray-project/xgboost_ray.git#egg=xgboost_ray
-git+https://github.com/ray-project/lightgbm_ray.git#lightgbm_ray
\ No newline at end of file
+git+https://github.com/ray-project/lightgbm_ray.git#lightgbm_ray
```
[the remark this answers is not in the document]
Comment (naijoaix) [reply release/requirements.txt:18]: Yeah they are identical, seems a bug?

File: dashboard/modules/serve/sdk.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (mwtian) [reply dashboard/modules/serve/sdk.py]: When we make another cut on Friday, I'm not sure if it will be 1.12 or 1.12rc1.

[the remark this answers is not in the document]
Comment (naijoaix) [reply dashboard/modules/serve/sdk.py]: I asked same question and I think we will use 1.12.0.rc1 @zhe-thoughts

File: release/requirements.txt (hunk 6 above)
[the remark this answers is not in the document]
Comment (mwtian) [reply release/requirements.txt:18]: Or different new line character ...? Anyway shouldn't be a big issue.

File: dashboard/modules/serve/sdk.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (mwtian) [reply dashboard/modules/serve/sdk.py]: Sounds good. It seems this should not matter for rc0, but in the final 1.12 candidate this probably should be 1.12.

File: release/requirements.txt (hunk 6 above)
[the remark this answers is not in the document]
Comment (naijoaix) [reply release/requirements.txt:18]: Didn't touch that...

**Comment by naijoaix** [comment]
Tests are passing. Any chance we can merge this?
